services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/AsyncTable.java

```diff
@@ -52,6 +53,9 @@
     /**
      * Retrieve the name of the DynamoDB table. This does not make a service call, and does not guarantee that the DynamoDB table
      * actually exists.
+     *
+     * <p>
+     * This call should never fail with a {@link RuntimeException}.
```

Comment:
This seems a bit ambiguous. Does it mean it might fail with a checked exception?

Reply:
I can fix this one pretty easily by specifying it will never throw an `Exception`, but what about elsewhere where it does sometimes throw `RuntimeException`s? Is that fine to leave as-is?

Reply:
You mean there are some places where checked exception might be thrown?

Reply:
Checked exceptions will never be thrown.

Reply:
Yeah, that's where it's a bit confusing, can we just say `This call should never throw a exception` so that people know right away without checking if it has `throws` to see if it throws a checked exception.

Reply:
Fixed!